pkg/pillar/devicenetwork/dnc.go

```diff
@@ -307,18 +312,35 @@ func VerifyPending(ctx *DeviceNetworkContext, pending *DPCPending,
 	return types.DPC_FAIL_WITH_IPANDDNS
 }
 
+type portError struct {
+	ifName string
+	err    error
+}
+
 // Check if all interfaces exist in the kernel
 // Returns the ifname for the first missing if there is an error
-func checkInterfacesExists(log *base.LogObject, dpc types.DevicePortConfig) (string, error) {
+func checkInterfacesExists(log *base.LogObject, dpc types.DevicePortConfig) ([]portError, types.DevicePortConfig) {
+	runnableDPC := types.DevicePortConfig{}
+	runnableDPC.Version = dpc.Version
+	runnableDPC.Key = dpc.Key
+	runnableDPC.TimePriority = dpc.TimePriority
+	runnableDPC.State = types.DPC_NONE
+	runnableDPC.TestResults = dpc.TestResults
+	runnableDPC.LastIPAndDNS = dpc.LastIPAndDNS
```

Comment:
In case there are other fields added, it would be easier to set runnable := dpc; runnable.Ports = nil
Don't know if need to clear other fields, but since MostlyEqual might compare them I suspect clearing Ports is all you want.

Reply:
makes sense.